smart-contracts/slither.db.json

```diff
@@ -0,0 +1 @@
 6}, "type_specific_fields": {"parent": {"type": "contract", "name": "CosmosBridge", "source_mapping": {"start": 223, "length": 8809, "filename_used": "/home/edd/Workspace/Sifchain/slither-playground/smart-contracts/contracts/CosmosBridge.sol", "filename_relative": "contracts/CosmosBridge.sol", "filename_absolute": "/home/edd/Workspace/Sifchain/slither-playground/smart-contracts/contracts/CosmosBridge.sol", "filename_short": "CosmosBridge.sol", "is_dependency": false, "lines": [10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 143, 144, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155, 156, 157, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168, 169, 170, 171, 172, 173, 174, 175, 176, 177, 178, 179, 180, 181, 182, 183, 184, 185, 186, 187, 188, 189, 190, 191, 192, 193, 194, 195, 196, 197, 198, 199, 200, 201, 202, 203, 204, 205, 206, 207, 208, 209, 210, 211, 212, 213, 214, 215, 216, 217, 218, 219, 220, 221, 222, 223, 224, 225, 226, 227, 228, 229, 230, 231, 232, 233, 234, 235, 236, 237, 238, 239, 240, 241, 242, 243, 244, 245, 246, 247, 248, 249, 250, 251, 252, 253, 254, 255, 256, 257, 258, 259, 260, 261, 262, 263, 264, 265, 266, 267, 268, 269, 270, 271, 272, 273, 274, 275, 276, 277, 278, 279, 280, 281, 282, 283, 284, 285, 286, 287, 288, 289, 290, 291, 292, 293, 294], "starting_column": 1, "ending_column": 2}}, "signature": "newProphecyClaim(CosmosBridgeStorage.ClaimType,bytes,uint256,address,string,uint256)"}}}, "additional_fields": {"underlying_type": "variables_written", "variable_name": "usedNonce"}}], "description": "Reentrancy in CosmosBridge.newProphecyClaim(CosmosBridgeStorage.ClaimType,bytes,uint256,address,string,uint256) (CosmosBridge.sol#129-199):\n\tExternal calls:\n\t- tokenAddress = BridgeBank(bridgeBank).createNewBridgeToken(symbol) (CosmosBridge.sol#156)\n\tState variables written after the call(s):\n\t- usedNonce[_prophecyID] = true (CosmosBridge.sol#192)\n", "markdown": "Reentrancy in [CosmosBridge.newProphecyClaim(CosmosBridgeStorage.ClaimType,bytes,uint256,address,string,uint256)](contracts/CosmosBridge.sol#L129-L199):\n\tExternal calls:\n\t- [tokenAddress = BridgeBank(bridgeBank).createNewBridgeToken(symbol)](contracts/CosmosBridge.sol#L156)\n\tState variables written after the call(s):\n\t- [usedNonce[_prophecyID] = true](contracts/CosmosBridge.sol#L192)\n", "id": "33c55e5c1cc291dc026c98eb1ac35d289902b1c75dddd09e36b239e009896209", "check": "reentrancy-no-eth", "impact": "Medium", "confidence": "Medium"}]
```

Comment:
Filename used should not have paths hard coded that are specifically for your computer e.g. : 

` "filename_used": "/home/edd/Workspace/Sifchain/slither-playground/smart-contracts/contracts/BridgeBank/BankStorage.sol"`

Let's use relative paths if we can.

Reply:
This is auto-generated from `slither . --triage` and then manually excluding each error as they arise. Because slither works by using the AST of a script they don't have any sort of `//ignore` style comments to ignore particular functions. Off the top of my head I can see two solutions here:

1. We have a bash script that replaces the hardcoded file paths to a string of your choice and add that to the `Makefile`
2. We have a custom config file (probably the function name and error to skip) and then parse the output of Slither to remove them before printing. 

Looks like 2 is how another team ended up solving this when using ConsenSys' Mythril tool.

Reply:
Ends up they're not required so manually deleted them - that was easier than expected :D Ready to be re-reviewed now.